java/kafka/consumer/src/main/java/KafkaConsumerExample.java

```diff
@@ -30,11 +45,21 @@ public static void main(String[] args) {
         Properties props = KafkaConsumerConfig.createProperties(config);
         int receivedMsgs = 0;
 
-        if (System.getenv("JAEGER_SERVICE_NAME") != null)   {
-            Tracer tracer = Configuration.fromEnv().getTracer();
-            GlobalTracer.registerIfAbsent(tracer);
+        TracingSystem tracingSystem;
+        tracingSystem = TracingSystem.forValue(config.getTracingSystem());
+
+        if (tracingSystem != null) {
+            if (tracingSystem == KafkaConsumerExample.TracingSystem.JAEGER) {
+                Tracer tracer = Configuration.fromEnv().getTracer();
+                GlobalTracer.registerIfAbsent(tracer);
 
-            props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, TracingConsumerInterceptor.class.getName());
+                props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, io.opentracing.contrib.kafka.TracingConsumerInterceptor.class.getName());
+            } else if (tracingSystem == KafkaConsumerExample.TracingSystem.OPENTELEMETRY) {
+
+                props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, io.opentelemetry.instrumentation.kafkaclients.TracingConsumerInterceptor.class.getName());
```

Comment:
When I compare it with @ppatierno and his PR in Strimzi operators ... he seems to do following in addition to configuring the interceptors:

```java
            System.setProperty("otel.metrics.exporter", "none"); // disable metrics
            AutoConfiguredOpenTelemetrySdk.initialize();
```

Is this needed here as well?

Reply:
Above lines for initialising the SDK now added.